Yoast/wordpress-seo
# P3 51 connect redux to component

## Context


* In this code we connect the first and following redux actions to the components:

1. Modal-opened REDUX ACTION, which opens the modal
2. Modal-dismissed REDUX ACTION, which corresponds to the dismissal of the modal
3. Change-database REDUX ACTION, which corresponds to changing the country of the database in the dropdown menu, in the related-keyphrases component ( P3-5: Create SEMrush Related-Keyphrases ComponentBLOCKED  )
4. New-request REDUX ACTION, which corresponds to the functionality of using the “Change country“ button in the related-keyphrases component ( P3-5: Create SEMrush Related-Keyphrases ComponentBLOCKED  ), after which a new request is made with the new database

* Furthermore we add some architectural changes to the containers which are discussed with @herregroen and @jcomack 

## Summary


This PR can be summarized in the following changelog entry:

* non-user facing

## Relevant technical choices:

* The containers are restructured and split up - The modal actions are dispatched in a separate container from the semrush container

## Test instructions

This PR can be tested by following these steps:

* Checkout this branch and build
* Make sure to have redux devtools installed
* Enter a keyphrase and hit the get related keyphrases button
* Make sure the MODAL_OPEN action gets called from 'yoast-seo/editor' within the redux dev tools and make sure that the expected values appear in the state
* Ensure the same for MODAL_CLOSE when closing the modal and make sure that the expected values appear in the state
* Open the modal again and select a country form the select box
* Ensure that the CHANGE_DATABASE action is called and make sure that the expected values appear in the state
* Close the modal and reopen and verify that it still contains the chosen country as preset
* Hit the change country button and verify that the NEW_REQUEST button is hit and make sure that the expected values appear in the state

## UI changes

## Quality assurance
